### ISSUE TITLE
Node can't find globally installed packages

### ISSUE THREAD

**Issue by jeff-kilbride** [coreybutler/nvm-windows#424]
If this is a question about how to use NVM4W, please use [stackoverflow](http://stackoverflow.com/questions/tagged/nvm-windows) instead.

If this is an issue regarding antivirus, make sure you search the existing issues first.

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL - see wiki for details)
- [ ] Windows 8
- [ ] Windows 8.1
- [x] Windows 10
- [ ] Windows 10 IoT Core
- [ ] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

- [ ] My Windows installation is non-English.

### I'm using NVM4W version:

- [x] 1.1.7
- [ ] 1.1.6
- [ ] 1.1.5
- [ ] 1.1.4
- [ ] 1.1.3
- [ ] 1.1.2
- [ ] 1.1.1
- [ ] Older
- [ ] OTHER (Please Specify)

### I have already...

- [x] read the README to be aware of npm gotchas & antivirus issues.
- [x] reviewed the [wiki](https://github.com/coreybutler/nvm-windows/wiki) to make sure my issue hasn't already been resolved.
- [x] verified I'm using an account with administrative privileges.
- [x] searched the issues (open and closed) to make sure this isn't a duplicate.
- [x] made sure this isn't a question about how to use NVM for Windows, since [gitter](https://gitter.im/coreybutler/nvm-windows?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) is used for questions and comments.

### My issue is related to (check only those which apply):

- [ ] settings.txt
- [ ] proxy support (Have you tried version 1.1.0+?)
- [ ] 32 or 64 bit support (Have you tried version 1.1.3+?)
- [ ] Character escaping (Have you tried version 1.1.6+?)
- [x] A standard shell environment (terminal/powershell)
- [x] A non-standard shell environment (Cmder, Hyper, Cygwin, git)

### Expected Behavior

Running a script using `node` from the command line should work with globally installed `npm` modules.

### Actual Behavior

`Error: Cannot find module xxxxxxx` from scripts as well as the `node` REPL for any globally installed modules.

### Steps to reproduce the problem:

Hi All,

I installed `nvm-windows` v1.1.7 along with `node` v10.15.1. I was even able to upgrade `npm` to v6.8.0 following the workarounds in other issues. Installing packages both globally and per project works as expected, and the global packages are installed to `C:\Program Files\nodejs\node_modules`. The `npm` cli works correctly and recognizes the packages:

```
C:\
λ nvm version
1.1.7

C:\
λ node --version
v10.15.1

C:\
λ npm --version
6.8.0

C:\
λ npm -g ls --depth=0
C:\Program Files\nodejs
+-- eslint@5.14.0
+-- eslint-config-airbnb-base@13.1.0
+-- eslint-config-prettier@4.0.0
+-- eslint-plugin-import@2.16.0
+-- eslint-plugin-prettier@3.0.1
+-- neat-csv@4.0.0
+-- npm@6.8.0
`-- prettier@1.16.4
```

However, if I try to `require` any of these packages in a script or using the `node` REPL, I get the following errors:

```
λ node convert.js                                                                        
internal/modules/cjs/loader.js:583                                                       
    throw err;                                                                           
    ^                                                                                    
                                                                                         
Error: Cannot find module 'neat-csv'                                                     
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)          
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)                     
    at Module.require (internal/modules/cjs/loader.js:637:17)                            
    at require (internal/modules/cjs/helpers.js:22:18)                                   
    at Object.<anonymous> (C:\Users\Jeff\Documents\VSCode\Trade Log\convert.js:6:13)     
    at Module._compile (internal/modules/cjs/loader.js:689:30)                           
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)             
    at Module.load (internal/modules/cjs/loader.js:599:32)                               
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)                             
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)                      
```
```
λ node                                                                                
> const Linter = require('eslint').Linter;                                            
{ Error: Cannot find module 'eslint'                                                  
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)       
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)                  
    at Module.require (internal/modules/cjs/loader.js:637:17)                         
    at require (internal/modules/cjs/helpers.js:22:18) code: 'MODULE_NOT_FOUND' }     
>
```

Even stranger -- I have the `eslint` and `prettier` extensions installed for `vscode`, and they both work using the global packages.

Both NVM_HOME and NVM_SYMLINK are defined in my environment variables and there is a correct `settings.txt` file in NVM_HOME. Everything else works great! For whatever reason, `node` can't find the global packages. If I install the package in the project directory, it works.

Am I missing something? This is a vanilla install to the default locations. I have tried running the script and REPL in both the Windows Terminal and Cmder.

Any help would be appreciated!

**Comment by repoman-git** [comment]
I have the same issue

**Comment by coreybutler** [comment]
This sounds less like an NVM4W issue, and more of a "how to use npm" issue.

Global modules aren't accessible within a project unless you install them locally. For example, you can't run `npm i -g somemodule` and then in your code use `require('somemodule')`. Node scripts do not look for global modules. They only look in the local project root and up the directory chain. Global modules are typically for installing CLI utilities or running npm scripts.

Instead, just run `npm i somemodule` in your project root, then you can `require` it in your script.

**Comment by repoman-git** [comment]
Hi @coreybutler, 
Thanks for the response.  It may well be a  "how to use npm" as I had never touched node/npm until a few weeks ago, but I believe that I followed the installation instructions to the letter.
 Prior to installation I uninstalled NPM, deleted 'program files\nodejs', '\appdata\npm', all the path entries and any other reference to node stuff I could find and deleted 'node_modules' folder in  the project.

I installed NVM,installed NPM, and installed 'yarn' attempting , however, thinking about it as I read your message I can't say hand on heart that a) I didn't use 'npm i -g' or that I didn't make a mistake.

I'll go over it again today.  Have a good day
